cli/lib.rs

```diff
@@ -344,6 +345,60 @@ async fn bundle_command(
   bundle_result
 }
 
+async fn doc_command(
+  flags: Flags,
+  source_file: String,
+  json: bool,
+  maybe_filter: Option<String>,
+) -> Result<(), ErrBox> {
+  let global_state = GlobalState::new(flags.clone())?;
+  let module_specifier =
+    ModuleSpecifier::resolve_url_or_path(&source_file).unwrap();
+  let source_file = global_state
+    .file_fetcher
+    .fetch_source_file(&module_specifier, None)
+    .await?;
+  let source_code = String::from_utf8(source_file.source_code)?;
+
+  let doc_parser = doc::DocParser::default();
+  let parse_result =
+    doc_parser.parse(module_specifier.to_string(), source_code);
+
+  let doc_nodes = match parse_result {
+    Ok(nodes) => nodes,
+    Err(e) => {
+      eprintln!("Failed to parse documentation:");
+      for diagnostic in e {
+        eprintln!("{}", diagnostic.message());
+      }
+
+      std::process::exit(1);
+    }
+  };
+
+  if json {
+    let json_string = serde_json::to_string_pretty(&doc_nodes).unwrap();
+    println!("{}", json_string);
+    return Ok(());
+  }
+
+  let printer = doc::TerminalPrinter::new();
```

Comment:
Rename and change behavior to output formatted string instead of actually printing in `TerminalPrinter`; handle SIGPIPE as above

Reply:
Add minimal tests for expected string output

Reply:
I fully agree with this. I have plans to rewrite how the printer works when I do the colorization (unless someone else wants to do it). I think it would be better if it returned a tree of tokens that can be rendered in color or without into a single string.